### PR TITLE
Add Float8E4M3B11FNUZ type support.

### PR DESCRIPTION
**Pull request by majnemer**
As proposed in [RFC: E4M3B11FNUZ in XLA](https://github.com/openxla/stablehlo/blob/main/rfcs/20230309-e4m3b11.md) (#1308), this change adds support for these types to StableHLO.

This includes the type definitions, vhlo, and interpreter support. The testing approach mirrors the Float8E4M3FNUZ tests, since it is also a "non-standard" floating point type supported by StableHLO.

### CODE REVIEW

File: stablehlo/tests/stablehlo_legalize_to_vhlo.0_11_0.mlir [hunk 1]
```diff
@@ -0,0 +1,2261 @@
...
+// CHECK-LABEL: "type_f8E5M2"
+
+func.func @type_f8E4M3FNUZ(%arg0: tensor<f8E4M3FNUZ>, %arg1: tensor<f8E4M3FNUZ>) -> tensor<f8E4M3FNUZ> {
+  // CHECK: "vhlo.add_v1"(%arg0, %arg1) : (!vhlo.tensor_v1<!vhlo.f8E4M3FNUZ_v1>, !vhlo.tensor_v1<!vhlo.f8E4M3FNUZ_v1>) -> !vhlo.tensor_v1<!vhlo.f8E4M3FNUZ_v1>
+  %0 = "stablehlo.add"(%arg0, %arg1) : (tensor<f8E4M3FNUZ>, tensor<f8E4M3FNUZ>) -> tensor<f8E4M3FNUZ>
+  func.return %0 : tensor<f8E4M3FNUZ>
+}
+// CHECK-LABEL: "type_f8E4M3FNUZ"
+
+func.func @type_f8E4M3B11FNUZ(%arg0: tensor<f8E4M3B11FNUZ>, %arg1: tensor<f8E4M3B11FNUZ>) -> tensor<f8E4M3B11FNUZ> {
```
Comment (GleasonK) [inline stablehlo/tests/stablehlo_legalize_to_vhlo.0_11_0.mlir:2161]: Could you add this test to `stablehlo_legalize_to_vhlo.mlir` as well? (https://github.com/openxla/stablehlo/blob/main/stablehlo/tests/stablehlo_legalize_to_vhlo.mlir)

Makes it easier to review what the new positive test is and keeps [VHLO change instructions](https://github.com/GleasonK/stablehlo/blob/main/docs/vhlo.md#add-versioned-serialization-test) simple for the next change to the opset.

File: stablehlo/tests/stablehlo_legalize_to_vhlo.0_11_0.mlir [hunk 2]
```diff
@@ -0,0 +1,2261 @@
+// RUN: stablehlo-opt --mlir-print-op-generic %s.bc | FileCheck %s
```
Comment (GleasonK) [inline stablehlo/tests/stablehlo_legalize_to_vhlo.0_11_0.mlir:1]: Could you add a negative test like the one here to ensure that this FP type cannot be serialized for v0.10.0:
https://github.com/openxla/stablehlo/blob/main/stablehlo/tests/vhlo_to_version_downgrade_invalid.0_9_0.mlir

**Review by GleasonK — APPROVED** [review]
This LGTM, two test case comments. Also now that I'm seeing that the type changes are in `Base.td`, the MHLO integration work may be taken care of already. Any negative test case changes (update expected error message), will be taken care of by the StableHLO oncall.

Will let @sdasgup3 review the `reference/` changes.

**Comment by GleasonK** [comment]
Also regarding the merge conflict, you can overwrite with your changes to `Version.h`

**Review by ghpvnist — APPROVED** [review]
LGTM on `spec.md` and `reference/` changes.

File: docs/spec.md [hunk 3]
```diff
@@ -250,6 +250,8 @@ values of type `tensor<T>`).
   * `f8E4M3FNUZ` and `f8E5M2FNUZ` types corresponding to the `E4M3` and `E5M2`
     encodings of the FP8 formats described in
     [8-bit Numerical Formats for Deep Neural Networks](https://arxiv.org/abs/2206.02915).
+  * `f8E4M3B11FNUZ` type corresponding to the `E4M3` encoding of the FP8 formats described in
```
Comment (GleasonK) [inline docs/spec.md:253]: https://github.com/openxla/stablehlo/actions/runs/4855239494/jobs/8654920963?pr=1448

Markdown lint check failing on this line, longer than 80chars

Comment (majnemer) [reply docs/spec.md]: I reflowed the line, how can I rerun linting?

Comment (burmako) [reply docs/spec.md]: The current approach involves having a committer running it for you, which I did :)

**Pull request merged**